enterprise/api/v1/views/enterprise_group.py

```diff
@@ -358,33 +385,28 @@ def remove_learners(self, request, group_uuid):
         param_serializer.is_valid(raise_exception=True)
 
         catalog_uuid = param_serializer.validated_data.get('catalog_uuid')
-        learner_emails = param_serializer.validated_data.get('learner_emails')
+        learner_emails = param_serializer.validated_data.get('learner_emails', [])
+        remove_all = param_serializer.validated_data.get('remove_all')
+
+        if bool(remove_all) == bool(learner_emails):
+            return Response("Must supply `remove_all` or `learner_email` but not both", status=400)
 
         records_deleted = 0
-        for user_email_batch in utils.batch(learner_emails[: 1000], batch_size=200):
-            existing_users = User.objects.filter(email__in=user_email_batch).values_list("id", flat=True)
-            group_q = Q(group=group)
-            ecu_in_q = Q(enterprise_customer_user__user_id__in=existing_users)
-            pecu_in_q = Q(pending_enterprise_customer_user__user_email__in=user_email_batch)
-            records_to_delete = models.EnterpriseGroupMembership.objects.filter(
-                group_q & (ecu_in_q | pecu_in_q),
-            )
+        if remove_all:
+            records_to_delete = models.EnterpriseGroupMembership.available_objects.filter(group=group)
             records_deleted += len(records_to_delete)
-            records_to_delete_uuids = [record.uuid for record in records_to_delete]
-            records_to_delete.delete()
-            for records_to_delete_uuids_batch in utils.batch(records_to_delete_uuids, batch_size=200):
-                send_group_membership_removal_notification.delay(
-                    customer.uuid,
-                    records_to_delete_uuids_batch,
-                    catalog_uuid)
-            # Woohoo! Records removed! Now to update the soft deleted records
-            deleted_records = models.EnterpriseGroupMembership.all_objects.filter(
-                uuid__in=records_to_delete_uuids,
-            )
-            deleted_records.update(
-                status=constants.GROUP_MEMBERSHIP_REMOVED_STATUS,
-                removed_at=localized_utcnow()
-            )
+            remove_records(records_to_delete, customer, catalog_uuid)
+        else:
+            for user_email_batch in utils.batch(learner_emails[: 1000], batch_size=200):
+                existing_users = User.objects.filter(email__in=user_email_batch).values_list("id", flat=True)
+                group_q = Q(group=group)
+                ecu_in_q = Q(enterprise_customer_user__user_id__in=existing_users)
+                pecu_in_q = Q(pending_enterprise_customer_user__user_email__in=user_email_batch)
+                records_to_delete = models.EnterpriseGroupMembership.available_objects.filter(
+                    group_q & (ecu_in_q | pecu_in_q),
+                )
```

Comment:
nit: The operation of getting all enterprise group memberships seems like it would be used in multiple places, maybe a helper method is in order?

Reply:
To me it feels as though this particular endpoint has a duplicate fetch. I think maybe I can just be smarter about my `remove_records` method